[2025-10-18T09:54:58.848+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-18T09:54:58.912+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wiki_trending_pipeline.clean_top_pages manual__2025-10-18T09:49:51.845181+00:00 [queued]>
[2025-10-18T09:54:58.924+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wiki_trending_pipeline.clean_top_pages manual__2025-10-18T09:49:51.845181+00:00 [queued]>
[2025-10-18T09:54:58.925+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2025-10-18T09:54:58.941+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): clean_top_pages> on 2025-10-18 09:49:51.845181+00:00
[2025-10-18T09:54:58.959+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=188) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-10-18T09:54:58.958+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'wiki_trending_pipeline', 'clean_top_pages', 'manual__2025-10-18T09:49:51.845181+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/wiki_dag/wiki_top_pages_dag.py', '--cfg-path', '/tmp/tmpk3qno5k6']
[2025-10-18T09:54:58.960+0000] {standard_task_runner.py:63} INFO - Started process 189 to run task
[2025-10-18T09:54:58.960+0000] {standard_task_runner.py:91} INFO - Job 6: Subtask clean_top_pages
[2025-10-18T09:54:58.982+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-10-18T09:54:59.037+0000] {task_command.py:426} INFO - Running <TaskInstance: wiki_trending_pipeline.clean_top_pages manual__2025-10-18T09:49:51.845181+00:00 [running]> on host 6ab8c125868b
[2025-10-18T09:54:59.182+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wiki_trending_pipeline' AIRFLOW_CTX_TASK_ID='clean_top_pages' AIRFLOW_CTX_EXECUTION_DATE='2025-10-18T09:49:51.845181+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-18T09:49:51.845181+00:00'
[2025-10-18T09:54:59.183+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-18T09:54:59.367+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-18T09:54:59.369+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/tasks/clean_top_pages.py", line 18, in clean_top_pages
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 497, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 515, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 107, in launch_gateway
    raise PySparkRuntimeError(
pyspark.errors.exceptions.base.PySparkRuntimeError: [JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
[2025-10-18T09:54:59.396+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=wiki_trending_pipeline, task_id=clean_top_pages, execution_date=20251018T094951, start_date=20251018T095458, end_date=20251018T095459
[2025-10-18T09:54:59.426+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 6 for task clean_top_pages ([JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.; 189)
[2025-10-18T09:54:59.455+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-10-18T09:54:59.527+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-18T09:54:59.529+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
